packages/block-library/src/navigation/index.php

```diff
@@ -710,3 +710,667 @@ function block_core_navigation_get_most_recently_published_navigation() {
 
 	return null;
 }
+
+/**
+ * WP_Navigation_Block_Renderer class
+ *
+ * @package gutenberg
+ * @since 6.5.0
+ */
+
+if ( class_exists( 'WP_Navigation_Block_Renderer' ) ) {
+	return;
+}
+
+/**
+ * Helper functions used to render the navigation block.
+ */
+class WP_Navigation_Block_Renderer {
```

Comment:
Is it possible to avoid a class and just have functions? The reason is more pragmatic than anything. We have code in place that auto rename functions in the "block-library" php files when building the Gutenberg plugin to avoid conflicts with core when the same files get included in Core.

I believe (to confirm) we don't have the same behavior for classes. So either we remove the class instead of regular functions or we update the webpack config that rename functions to also prefix classnames.

Reply:
The reason for using a class is that we can keep keep the public functions to a minimum, which helps to avoid creating a load of functions we have to maintain for BC.

Reply:
I don't feel strongly either. There are already many public functions so I'm not sure making these public loads worse!

Reply:
Have we investigated how hard it would be to add the auto-renaming functionality? As Riad says we're unopinionated about the code so we should probably aim to do what's right for the code and make the release process conform to that.

That said, if we stick with class then any help looking into auto-renaming would be great @scruffian 🙏 😄